src/main/http-client.ts

```diff
@@ -62,20 +63,44 @@ function createMethod<R, P>(
             'content-type': 'application/json',
             ...config.headers,
         };
-        const res = await fetch(url.toString(), {
-            method,
-            headers,
-            mode: 'cors',
-            credentials: 'include',
-            body: method === 'POST' ? JSON.stringify(params ?? {}) : undefined,
-        });
-        const json = await res.json().catch(() => ({}));
-        if (!res.ok) {
-            const error = new Error(json?.message ?? 'Unknown error') as any;
-            error.name = json.name ?? 'UnknownError';
-            error.details = json.details;
-            throw error;
+        try {
+            const res = await fetch(url.toString(), {
+                method,
+                headers,
+                mode: 'cors',
+                credentials: 'include',
+                body: method === 'POST' ? JSON.stringify(params ?? {}) : undefined,
+            });
+
+            if (!res.ok) {
+                const responseBodyText = await res.text();
+                const json = parseJson(responseBodyText, {});
+                const message = json.message ?? responseBodyText;
+                const error = new Error(message) as any;
+                error.name = json.name ?? 'UnknownError';
+                error.details = json.details;
+                error.code = res.status;
+                throw error;
+            }
+            return await res.json();
```

Comment:
Since we're changing the flow here slightly, would probably be useful to handle non-json successful responses here (b/c those are also unexpected from protocomm backends). Currently it'll be reported as "RequestError" with "SyntaxError" cause, which might be ok, but still "blends" with errors thrown by `fetch` line above